public/app/plugins/datasource/prometheus/querybuilder/shared/operationUtils.ts

```diff
@@ -144,3 +148,169 @@ export function getRangeVectorParamDef(withRateInterval = false): QueryBuilderOp
+  operations[2].params.unshift(...paramsDef.params);
+  operations[0].defaultParams = paramsDef.defaultParams;
+  operations[1].defaultParams = [...paramsDef.defaultParams, ''];
+  operations[2].defaultParams = [...paramsDef.defaultParams, ''];
+  operations[1].renderer = getAggregationByRendererWithParameter(name);
+  operations[2].renderer = getAggregationByRendererWithParameter(name);
+  return operations;
+}
+
+function getAggregationByRenderer(aggregation: string) {
+  return function aggregationRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
+    return `${aggregation} by(${model.params.join(', ')}) (${innerExpr})`;
+  };
+}
+
+function getAggregationWithoutRenderer(aggregation: string) {
+  return function aggregationRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
+    return `${aggregation} without(${model.params.join(', ')}) (${innerExpr})`;
+  };
+}
+
+/**
+ * Very simple poc implementation, needs to be modified to support all aggregation operators
+ */
+function getAggregationExplainer(aggregationName: string, mode: 'by' | 'without' | '') {
+  return function aggregationExplainer(model: QueryBuilderOperation) {
+    const labels = model.params.map((label) => `\`${label}\``).join(' and ');
+    const labelWord = pluralize('label', model.params.length);
+
+    switch (mode) {
+      case 'by':
+        return `Calculates ${aggregationName} over dimensions while preserving ${labelWord} ${labels}.`;
+      case 'without':
+        return `Calculates ${aggregationName} over the dimensions ${labels}. All other labels are preserved.`;
+      default:
+        return `Calculates ${aggregationName} over the dimensions.`;
+    }
+  };
+}
+
+function getAggregationByRendererWithParameter(aggregation: string) {
+  return function aggregationRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
+    function mapType(p: QueryBuilderOperationParamValue) {
+      if (typeof p === 'string') {
+        return `\"${p}\"`;
+      }
+      return p;
+    }
+    const params = model.params.slice(0, -1);
+    const restParams = model.params.slice(1);
+    return `${aggregation} by(${restParams.join(', ')}) (${params.map(mapType).join(', ')}, ${innerExpr})`;
+  };
+}
+
+/**
+ * This function will transform operations without labels to their plan aggregation operation
+ */
+function getLastLabelRemovedHandler(changeToOperartionId: string) {
+  return function onParamChanged(index: number, op: QueryBuilderOperation, def: QueryBuilderOperationDef) {
+    // If definition has more params then is defined there are no optional rest params anymore
+    // We then transform this operation into a different one
+    if (op.params.length < def.params.length) {
+      return {
+        ...op,
+        id: changeToOperartionId,
+      };
+    }
+
+    return op;
+  };
+}
+
+function getOnLabelAdddedHandler(changeToOperartionId: string) {
```

Comment:
```suggestion
function getOnLabelAdddedHandler(changeToOperationId: string) {
```

replace all `changeToOperartionId` -> `changeToOperartionId `